modules/osdk-internal-crds.adoc

```diff
@@ -0,0 +1,51 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-generating-csvs.adoc
+
+[id="osdk-internal-crds_{context}"]
+= Internal or data-only objects
+
+It is a common practice for Operators to use Custom Resource Definitions (CRDs)
+internally to accomplish a task. These objects are not meant for users to
+manipulate and can be confusing to users of the Operator. For example, a
+database Operator might have a Replication CRD that is created whenever a user
+creates a Database object with `replication: true`.
+
+If a CRD is not meant for manipulation by users, it can be hidden by including
+its name within the `apps.operatorframework.io/internal-objects` annotation's
+array of values in the Operator's ClusterServiceVersion (CSV):
+
+.Internal object annotation
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: couchdb-operator-v1.2.3
+  annotations:
+    apps.operatorframework.io/internal-objects:
+      - '(Internal) CouchDB Formation Lock'
+      - '(Internal) CouchDB Recipe'
+[...]
+----
+
+If a CRD is only meant for tracking data, it can also be included within the
+`app.operatorframework.io/data-object` array:
+
+.Data-only object annotation
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: couchdb-operator-v1.2.3
+  annotations:
+    apps.operatorframework.io/data-object:
+      - '(Internal) CouchDB Formation Lock'
+      - '(Internal) CouchDB Recipe'
+[...]
+----
+
+Before marking one of your CRDs as internal, make sure that any debugging
+information or configuration that might be required to manage the application is
+reflected on the CR's status or `spec` block.
```

Comment:
Do we need an example of what this might look like, or is it my ignorance here?

Reply:
It's going to vary depending on the application, so I've added "if applicable to your Operator".